src/hotspot/share/runtime/os.hpp

```diff
@@ -610,7 +614,8 @@ class os: AllStatic {
                                               address addr,
                                               char* buf = NULL, int buflen = 0,
                                               bool shorten_paths = true,
-                                              bool demangle = true);
+                                              bool demangle = true,
+                                              bool strip_arguments = false);
```

Comment:
Are you actually relying on these having default values now?

Reply:
I think it makes sense. The scratch buffer arguments are useful during error handling but a bit cumbersome in normal situations. Since I want this to be a general purpose function, I make the scratch buffer optional.

Btw, do you object against the `alloca` in the implementation? We use it in some few other places too. But it was a coin toss really whether to use (raw) malloc instead.